test/08_api-keys.ts

```diff
@@ -181,6 +191,67 @@ describe('create device apikey', function () {
 				expect(deviceApiKey).to.equal(apiKey);
 			});
 
+			it('should create an apikey with only the key-name passed in the body', async function () {
+				const { body: deviceApiKey } = await fn(this.user, this.device.id, {
+					name: `SampleDeviceKey${i}`,
+				}).expect(200);
+
+				expect(deviceApiKey).to.be.a('string');
+				expect(deviceApiKey).to.not.be.empty;
+
+				// check the name:description assigned
+				const apiKeyResp = await api.resin.get({
+					resource: 'api_key',
+					passthrough: {
+						req: permissions.root,
+					},
+					id: {
+						key: deviceApiKey,
+					},
+					options: {
+						$select: ['name'],
+					},
+				});
+
+				expect(apiKeyResp).to.have.property('name', `SampleDeviceKey${i}`);
+			});
+
+			it('should create an apikey with the key-name and key-description passed in the body', async function () {
+				const { body: deviceApiKey } = await fn(this.user, this.device.id, {
+					name: `SampleDeviceKey${i}`,
+					description: `Sample key ${i} description.`,
+				}).expect(200);
+
+				expect(deviceApiKey).to.be.a('string');
+				expect(deviceApiKey).to.not.be.empty;
+
+				// check the name:description assigned
+				const apiKeyResp = await api.resin.get({
+					resource: 'api_key',
+					passthrough: {
+						req: permissions.root,
+					},
+					id: {
+						key: deviceApiKey,
+					},
+					options: {
+						$select: ['name', 'description'],
+					},
+				});
+
+				expect(apiKeyResp).to.have.property('name', `SampleDeviceKey${i}`);
+				expect(apiKeyResp).to.have.property(
+					'description',
+					`Sample key ${i} description.`,
+				);
+			});
+
+			it('should return BadRequest for an apikey create request where the key name is not string', async function () {
+				await fn(this.user, this.device.id, {
+					name: 123,
+				}).expect(400);
```

Comment:
It's nice to check the error message as well, as there's been times tests have been passing for the wrong reasons.

```suggestion
				}).expect(400, 'Key name should be a string value');
```